### PR TITLE
[PCM-2114] sdk side of session reinvites

### PR DESCRIPTION
**Pull request by jensengar**

### CODE REVIEW

File: src/sessions/base-session-handler.ts [hunk 1]
```diff
@@ -29,6 +29,10 @@ export default abstract class BaseSessionHandler {
 
   constructor (protected sdk: GenesysCloudWebrtcSdk, protected sessionManager: SessionManager) {
     this.sdk.on("sessionEnded", (session, reason) => {
+      if (session.sessionType !== this.sessionType) {
```
Comment (jensengar) [inline src/sessions/base-session-handler.ts:32]: This fixes the log with the `sessionEnded event` which was triggering for all session handlers, not just the one that handles the ended session.

File: src/sessions/base-session-handler.ts [hunk 2]
```diff
@@ -92,7 +96,10 @@ export default abstract class BaseSessionHandler {
     });
 
     session.on('terminated', this.onSessionTerminated.bind(this, session));
-    this.sdk.emit('sessionStarted', session);
+
+    if (!session.reinvite) {
```
Comment (jensengar) [inline src/sessions/base-session-handler.ts:100]: we want the reinvite sessions to silently replace the existing session so I can't think of a reason why we'd want to notify consumers about a reinvite session.

File: test/test-pages/common/sdk-controller.js [hunk 3]
```diff
@@ -213,16 +213,16 @@ function setConversationHeld ({ held, conversationId }) {
   webrtcSdk.setConversationHeld({ held, conversationId });
 }
 
-function endSession () {
-  webrtcSdk.endSession({ conversationId: currentConversationId });
+function endSession (props) {
+  webrtcSdk.endSession(props);
```
Comment (jensengar) [inline test/test-pages/common/sdk-controller.js:217]: this fixes the demo app

File: test/test-pages/common/sdk-controller.js [hunk 4]
```diff
@@ -213,16 +213,16 @@ function setConversationHeld ({ held, conversationId }) {
...
 
   document.getElementById('video-actions').classList.add('hidden');
   const startControls = document.querySelectorAll('.start-controls');
   startControls.forEach(el => el.classList.remove('hidden'));
 }
 
-function setAudioMute (mute) {
-  webrtcSdk.setAudioMute({ conversationId: currentConversationId, mute });
+function setAudioMute (props) {
+  webrtcSdk.setAudioMute(props);
```
Comment (jensengar) [inline test/test-pages/common/sdk-controller.js:225]: this fixes the demo app

File: package.json [hunk 5]
```diff
@@ -61,7 +61,7 @@
     "browserama": "^3.1.1",
     "core-js": "^3.7.0",
     "genesys-cloud-client-logger": "^4.2.0",
-    "genesys-cloud-streaming-client": "^15.1.3",
+    "genesys-cloud-streaming-client": "^16.0.1-develop.52",
```
Comment (zservies) [inline package.json:64]: I'm assuming this is just placeholder until the streaming-client is released?

**Review by zservies — APPROVED** [review]

**Review by maxwellmooney13 — APPROVED** [review]

**Review by hjon — APPROVED** [review]

File: test/test-pages/common/sdk-controller.js (hunk 4 above)
Comment (jensengar) [reply test/test-pages/common/sdk-controller.js]: yessir

**Pull request merged**